### PR TITLE
Merging after PISA repo moved to IceCubeOpenSource

### PR DESCRIPTION
**Pull request by ts4051**
First batch of changes to merge following the PISA repo move. These are changes that were made during the wait for the move.

Also one fix required to the current master branch is implemented (pint version issue).

Changes are pretty uncontroversial, e.g. a few minor fixes. Some major changes to the simple parameterised reco stage (used for Upgrade work). Is more general now, and more steerable from the cfg file. Docs updated accordingly.

### CODE REVIEW

**Comment by jllanfranchi** [comment]
"Also one fix required to the current master branch is implemented (pint version issue)." ... what exactly is the version issue? Does the fix require a particular version (in which case you need to update requirements.txt and setup.py)? Did it crop up between specific versions but the fix works regardless of version? This kind of information is important to communicate so we can avoid the issue you found in perpetuity.

**Comment by ts4051** [comment]
For the pint issue, the issue is only in newer versions it seems. I guess specifying an old version isn't great practise, which is why I added the current fix.

The issue can be shown simply using the following code snippet:

```
import collections
from pisa import ureg
import pint
print("pint version : %s" % pint.__version__)
a = 123. * ureg.dimensionless
print("%s is an Iterable : %s" % (type(a),isinstance(a,collections.Iterable)))
```
On my laptop (newer pint) the problem is seen, output below:
```
pint version : 0.9
<class 'pint.quantity.Quantity'> is an Iterable : False
```

On cobalt (older pint):
```
pint version : 0.8.1
<class 'pint.quantity.Quantity'> is an Iterable : True
```

File: pisa/stages/reco/simple_param.py [hunk 1]
```diff
@@ -40,7 +40,69 @@
  limitations under the License.'''
 
 
-def get_visible_energy(particle_key,true_energy) :
+def dict_lookup_wildcard(dict_obj,key) :
+
+    import fnmatch, collections
```
[the remark this answers is not in the document]
Comment (ts4051) [reply pisa/stages/reco/simple_param.py]: Fixed

File: pisa/stages/reco/simple_param.py [hunk 2]
```diff
@@ -40,7 +40,69 @@
  limitations under the License.'''
 
 
-def get_visible_energy(particle_key,true_energy) :
+def dict_lookup_wildcard(dict_obj,key) :
+
```
[the remark this answers is not in the document]
Comment (ts4051) [reply pisa/stages/reco/simple_param.py]: Fixed

**Comment by jllanfranchi** [comment]
So your fix works with either version of PInt that you tested?

**Comment by ts4051** [comment]
Correct, the fix supports any pint version I believe. E.g. it supports anything iterable as usual (in pint or otherwise), but now also supports Quantity arrays in pint versions where these aren't deemed iterable.

**Comment by ts4051** [comment]
I think everything is addressed here, let me know if I've missed something. Otherwise, please could we go ahead with the merge.

**Comment by ts4051** [comment]
Since this is still open, I've added too more small changes/fixes. Might as well get it all in at once! 
1) pipeline/stage.get_outputs() can now return the events (previously could only return a mapset)
2) toy event generator brought up-to-date with latest code

**Comment by jllanfranchi** [comment]
Everything looks in order to me, thanks for the fixes and improvements! I'll go ahead and merge.

**Comment by ts4051** [comment]
Great, thanks for the review

**Pull request merged**